### ISSUE TITLE
Intermittent slowness

### ISSUE THREAD

**Issue by burner1024** [WeblateOrg/weblate#1559]
### Steps to reproduce
I'm not sure. This started after upgrading Weblate to 2.15 and adding a new, larger project.

### Actual behaviour
A row of "save" requests take a minute each:
```
[pid: 27173|app: 0|req: 14/57] xx.xx.xxx.xx () {68 vars in 1546 bytes} [Mon Jul 17 19:49:24 2017] POST /translate/fallout2-up/male/es/?type=check%3Aend_ellipsis&offset=5 => generated 0 bytes in 61222 msecs (HTTP/1.1 302) 6 headers in 411 bytes (1 switches on core 0)
```
In top, I see Mysql spike to 200% CPU for a moment, then uwsgi keeps at 90-100% for the whole minute. Mysql slow logs don't show anything over 1 sec. I also tried disabling Redis cache, that didn't help.

Looks like a timeout of some sort, but I'm not sure how to debug it.

### Expected behaviour
A "save translation" request should not take a minute.

### Server configuration
```
 * Weblate weblate-2.15
 * Python 2.7.5
 * Django 1.10.3
 * six 1.10.0
 * social-auth-core 1.4.0
 * social-auth-app-django 1.2.0
 * django-appconf 1.0.2
 * Translate Toolkit 2.0.0
 * Whoosh 2.7.4
 * defusedxml 0.4.1
 * Git 2.13.2
 * Pillow (PIL) 1.1.7
 * dateutil 2.6.0
 * lxml 3.6.4
 * django-crispy-forms 1.6.1
 * compressor 2.1.1
 * djangorestframework 3.5.3
 * pytz 2016.7
 * pyuca N/A
 * python-bidi 0.4.0
 * pyLibravatar N/A
 * PyYAML 3.12
 * Mercurial 4.0
 * Database backends: django.db.backends.mysql
```

**Comment by nijel** [comment]
This has been most likely caused by not storing updated hash on save, it has been fixed shortly after 2.15 release in 66a7391e7bba9bc9ed22a9340fe673e888ecb804...

**Comment by burner1024** [comment]
Thank you for swift response. I have applied the fix manually, and it seems to have worked.

I will re-open the issue if it re-surfaces.
